src/components/Requirements/SubRequirement.vue

```diff
@@ -173,19 +173,24 @@ type Data = {
 
 const generateSubReqIncompleteCourses = (
   allTakenCourseIds: ReadonlySet<number>,
-  eligibleCourseIds: readonly number[]
-): readonly FirestoreSemesterCourse[] => {
+  eligibleCourseIds: readonly number[],
+  requirementID: number
+): readonly AppFirestoreSemesterCourseWithRequirementID[] => {
   const rosterCourses = eligibleCourseIds
     .filter(courseID => !allTakenCourseIds.has(courseID))
     .flatMap(courseID => fullCoursesJson[courseID] || []);
   const subjectColors = allocateSubjectColors(new Set(rosterCourses.map(it => it.subject)));
-  return rosterCourses.map(rosterCourse =>
+  const coursesWithDummyUniqueID = rosterCourses.map(rosterCourse =>
     cornellCourseRosterCourseToFirebaseSemesterCourseWithCustomIDAndColor(
       rosterCourse,
       -1,
       subjectColors[rosterCourse.subject]
     )
   );
+  return coursesWithDummyUniqueID.map(course => ({
+    ...course,
+    requirementID,
```

Comment:
I think this may be of type `number` not `string`. I believe `requirementID` was defined as having type `string` for `AppFirestoreSemesterCourseWithRequirementID`. I believe this is why the CI build is failing.